app/celer-g4/RunInput.cc

```diff
@@ -6,13 +6,173 @@
 //---------------------------------------------------------------------------//
 #include "RunInput.hh"
 
+#include <fstream>
+
+#include "corecel/Config.hh"
+
 #include "corecel/io/EnumStringMapper.hh"
+#include "corecel/io/Logger.hh"
+#include "corecel/math/ArrayUtils.hh"
+#include "celeritas/inp/StandaloneInput.hh"
+#include "celeritas/phys/PrimaryGeneratorOptions.hh"
 #include "accel/SharedParams.hh"
 
 namespace celeritas
 {
 namespace app
 {
+namespace
+{
+//---------------------------------------------------------------------------//
+inp::System load_system(RunInput const& ri)
+{
+    inp::System s;
+
+    if (celeritas::Device::num_devices())
+    {
+        inp::Device d;
+        d.stack_size = ri.cuda_stack_size;
+        d.heap_size = ri.cuda_heap_size;
+        s.device = std::move(d);
```

Comment:
Since all struct fields are being initialized, can just write
```
s.device = {
    .stack_size = ri.cuda_stack_size,
    .heap_size = ri.cuda_heap_size
};
```
or if you want the abbreviation, maybe
```
inp::Device& d = s.device;
```
?

Reply:
Neither will work because `device` is `std::optional`. The first usage also requires C++20 and we still want to support C++17.

Reply:
I think in c++20 we could preferably do
```c++
s.device = inp::Device{
    .stack_size = ri.cuda_stack_size,
    .heap_size = ri.cuda_heap_size
};
```

Reply:
In the meantime you have also use:
```suggestion
        s.device = inp::Device{};
        s.device.stack_size = ri.cuda_stack_size;
        s.device.heap_size = ri.cuda_heap_size;
```
or even (less descriptive I guess)
```
        s.device = Device{ri.cuda_stack_size, ri.cuda_heap_size};
```